itez/beneficiary/migrations/0007_auto_20211117_1306.py

```diff
@@ -0,0 +1,35 @@
+# Generated by Django 3.1.13 on 2021-11-17 13:06
+
+from django.db import migrations, models
+import django.db.models.deletion
+import uuid
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('beneficiary', '0006_auto_20211110_0939'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='workdetail',
+            name='beneficiary',
+            field=models.OneToOneField(blank=True, null=True, on_delete=django.db.models.deletion.PROTECT, to='beneficiary.beneficiary'),
+        ),
+        migrations.AlterField(
+            model_name='agentdetail',
+            name='agend_ID',
+            field=models.CharField(default='AF3EF2F', max_length=100, verbose_name='Agent Id'),
+        ),
+        migrations.AlterField(
+            model_name='beneficiary',
+            name='agent_ID',
+            field=models.ForeignKey(default='510B20C', on_delete=django.db.models.deletion.PROTECT, to='beneficiary.agentdetail'),
+        ),
+        migrations.AlterField(
+            model_name='beneficiary',
+            name='beneficiary_ID',
+            field=models.UUIDField(default=uuid.UUID('5942f8fc-9e94-4fb0-8f1d-276e3e3181ee'), editable=False),
```

Comment:
so for us to get the id we desire in the model of this
lets set this to  charfied and non editable 
and then set the beneficiary id  as default from the uuid function

Reply:
I have resolved this to use Django signals to generate a unique id.